client/my-sites/site-settings/form-base.js

```diff
@@ -131,7 +139,9 @@ module.exports = {
 
 		notices.clearNotices( 'notices' );
 
-		this.setState( { submittingForm: true } );
+		//for dirtyFields, see lib/mixins/dirty-linked-state
+		this.setState( { submittingForm: true, dirtyFields: [] } );
```

Comment:
Should we move the clearing of `dirtyFields` down to the success case below? If there's some transient internet hiccup and you're in the error case where you just need to resubmit, I think we'd still want the `dirtyFields` to be retained until the form is successfully submitted. Otherwise your changes would potentially be cleared out while you are in that error state.